presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergTableHandle.java

```diff
@@ -85,9 +89,20 @@ public TableType getTableType()
     }
 
     @JsonProperty
+    public TupleDomain<HiveColumnHandle> getEnforcedPredicate()
+    {
+        return enforcedPredicate;
+    }
+
+    @JsonProperty
+    public TupleDomain<HiveColumnHandle> getEffectivePredicate()
+    {
+        return effectivePredicate;
+    }
+
     public TupleDomain<HiveColumnHandle> getPredicate()
     {
-        return predicate;
+        return columnWiseUnion(enforcedPredicate, effectivePredicate);
```

Comment:
Should we use `TupleDomain::intersect()` here?

Reply:
Yeah. I missed it